src/main/java/pl/stormit/eduquiz/authenticator/controllers/AuthenticationViewController.java

```diff
@@ -0,0 +1,61 @@
+package pl.stormit.eduquiz.authenticator.controllers;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import pl.stormit.eduquiz.quizcreator.domain.user.UserService;
+import pl.stormit.eduquiz.quizcreator.domain.user.dto.UserRequestDto;
+
+@Controller
+@RequestMapping("/register")
+@AllArgsConstructor
+public class AuthenticationViewController {
+
+
+    private final UserService userService;
+
+    @GetMapping()
+    public String registerUser(final Model model, Error error) {
+        model.addAttribute("userRequestDto", new UserRequestDto(null,
+                null, null, null, null, null, null));
+        model.addAttribute("error", error);
+        return "register";
+    }
+
+    @PostMapping()
```

Comment:
this method can be made a little clearer in the following way ( of course how what I wrote will work :P)

@PostMapping("/register")
public String registerUser(@ModelAttribute UserRequestDto userRequestDto, BindingResult bindingResult, Model model) {

    if (!userService.comparePasswords(userRequestDto.getPassword(), userRequestDto.getConfirmPassword())) {
        bindingResult.rejectValue("confirmPassword", "userRequestDto.confirmPassword", "Passwords are not the same");
    }

    if (bindingResult.hasErrors()) {
        return "register";
    }

    if (userService.checkIfNicknameAvailable(userRequestDto)) {
        bindingResult.rejectValue("nickname", "userRequestDto.nickname", "An account already exists for this nickname.");
        return "register";
    }

    userService.createUser(userRequestDto);

    model.addAttribute("userRequestDto", new UserRequestDto(null, null, null, null, null, null, null));
    model.addAttribute("message", "User was successfully registered!");

    return "register";
}